enclave-runtime/src/lib.rs

```diff
@@ -251,19 +286,19 @@ pub unsafe extern "C" fn mock_register_enclave_xt(
 	)
 	.encode();
 
+	*nonce += 1;
+
 	write_slice_and_whitespace_pad(extrinsic_slice, xt);
 	sgx_status_t::SGX_SUCCESS
 }
 
-fn create_extrinsics<PB, V>(
-	validator: &V,
+fn create_extrinsics<PB>(
+	genesis_hash: HashFor<PB>,
 	calls: Vec<OpaqueCall>,
-	mut nonce: u32,
+	nonce: &mut u32,
```

Comment:
Also here I think it would probably make sense to have an abstraction layer to increase the nonce, rather than access to the raw value that we increment.

Reply:
Yes, I think so too, but I just did not want to introduce other rather unrelated changes in this PR. Thanks to your input I just had the idea of an extrinsic factory inside the enclave. This might be suitable abstraction. I have updated the description of #429.

Reply:
Alright, that's fine for me 👍